### ISSUE TITLE
Disabling video doesn't do videoStream.stop()

### ISSUE THREAD

**Issue by JohnMcLear** [ether/ep_webrtc#22]
This means that video is still captured just not published to screen or recipients.  For security and privacy reasons we should ensure .stop is fired.

**Comment by orblivion** [comment]
I'm curious if you're referring to the "Enable Audio/Video Chat" button, or the video "enable/disable" toggle next to the "mute/unmute" button.

If you're referring to the first, PR is #34 

If you're referring to the second, I wonder if it's the right approach. From the docs, it seems that `enabled` seems to be for "muting" both video and audio. `stop()` should be when you're done. I don't think you can "un-stop". To re-enable video, I think you'd have to create a new stream. At least that's my suspicion. If this is what you had in mind for me to do, I can investigate further to confirm.

**Comment by JohnMcLear** [comment]
Whatever feels natural/right to you just roll with that :)

**Comment by orblivion** [comment]
Hmm weird it might be platform dependent. It turns off for me, at least it is now.

**Comment by orblivion** [comment]
Ah, so Chromium does not turn off the camera.

**Comment by orblivion** [comment]
Here's a bug report to Chromium on this issue, comments ranging from 2016 to late 2019: https://bugs.chromium.org/p/chromium/issues/detail?id=642785 Seems to echo my observations and impressions of the purposes of these modes. Sounds like it's not just us, and it's an open question in Chromium land.

It seems that with Chromium, you currently have the choice of using `enabled = false` which leaves the light on when video is "muted", or using `stop()` which may prompt the user for permission again which is perhaps not the UI you want here for a video "mute" button.

**Comment by JohnMcLear** [comment]
Seems to be fine in FF on Windows.

Chrome exhibits this "bug" on Windows.

I don't mind using stop tho, for Chrome at least, because once you accept perms and click to leave enabled it wont prompt again.  I think we should go for the most privacy focused option right now but maybe put it under a setting so a site admin can decide the level of privacy to push.

**Comment by orblivion** [comment]
It just occurred to me as I work on this - If reenabling video means resetting the connection, the users' audio will be disrupted, right?

**Comment by orblivion** [comment]
@JohnMcLear and I discussed off-channel. We'd like to put this off until the browsers settle on a standard for the behavior of camera/mic permission asking, and for the camera light while `enabled = false`. Currently, Firefox and Chromium behave differently (not sure about other browsers).

The reason the fix would be costly is that `videoTrack.stop` is one-way. There's no `videoTrack.restart`. (Unlike `enabled = true` vs `enabled = false`). This means that if we used `videoTrack.stop` for Chrome users instead of `enabled = false`, we would need to restart the entire stream to turn the video back on, acquiring a new `videoTrack` and `audioTrack`. The problems with these are:

* This introduces the possibility of interrupting the audio stream
* Requires us to save the state of the previous stream (audio muted? video expanded?)
    * IMHO this introduces complications and possible bugs into the flow. And/or requires a restructuring of a fair amount of code.
